pilot/pkg/authn/authentication.go

```diff
@@ -12,19 +12,17 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package model
+package authn
```

Comment:
I'm not sure if moving this out of the model directory aligns with the code structure in pilot, but it might be just fine. cc @howardjohn @costinm 

But regarding the new directory `pilot/pkg/authn/`, I don't feel it's the right place to put this, especially considering we already have the `pilot/pkg/security/authn`

cc @diemtvu

Reply:
+1 on moving to `pilot/pkg/security`. May be `pilot/pkg/security/model` to preserve the semantic.